kernel: bump 6.1 to 6.1.57

graysky2 opened the pull request:
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.1.57
```
Manually rebased:
	generic/pending-6.1/702-net-ethernet-mtk_eth_soc-enable-threaded-NAPI.patch

Removed upstreamed:
	qualcommax/patches-6.1/0134-PCI-qcom-Fixing-broken-pcie-enumeration-for-2_3_3-co.patch[1]
```
All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.57&id=2dfb5f324d799f4545e17631415aba6d302a8e2b

Build system: x86/64
Build-tested: x86/64/AMD Cezanne
Run-tested: x86/64/AMD Cezanne

graysky2 committed:
kernel: bump 6.1 to 6.1.57

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.1.57

Manually rebased:
	generic/pending-6.1/702-net-ethernet-mtk_eth_soc-enable-threaded-NAPI.patch

Removed upstreamed:
	qualcommax/patches-6.1/0134-PCI-qcom-Fixing-broken-pcie-enumeration-for-2_3_3-co.patch[1]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.57&id=2dfb5f324d799f4545e17631415aba6d302a8e2b

Build system: x86/64
Build-tested: x86/64/AMD Cezanne
Run-tested: x86/64/AMD Cezanne

Reviewed-by: Robert Marko <robimarko@gmail.com>
Signed-off-by: John Audia <therealgraysky@proton.me>

Ansuel commented on target/linux/generic/pending-6.1/702-net-ethernet-mtk_eth_soc-enable-threaded-NAPI.patch:
```diff
@@ -10,26 +10,6 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 
 --- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
 +++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
-@@ -3155,8 +3155,8 @@ static irqreturn_t mtk_handle_irq_rx(int
- 
- 	eth->rx_events++;
- 	if (likely(napi_schedule_prep(&eth->rx_napi))) {
--		__napi_schedule(&eth->rx_napi);
- 		mtk_rx_irq_disable(eth, eth->soc->txrx.rx_irq_done_mask);
-+		__napi_schedule(&eth->rx_napi);
- 	}
- 
- 	return IRQ_HANDLED;
-@@ -3168,8 +3168,8 @@ static irqreturn_t mtk_handle_irq_tx(int
- 
- 	eth->tx_events++;
- 	if (likely(napi_schedule_prep(&eth->tx_napi))) {
--		__napi_schedule(&eth->tx_napi);
- 		mtk_tx_irq_disable(eth, MTK_TX_DONE_INT);
-+		__napi_schedule(&eth->tx_napi);
- 	}
- 
- 	return IRQ_HANDLED;
```
LOL... no idea we had this change... I accidentally pushed the same change while reworking napi_schedule and noticing the problem in how it was handled.

graysky2 replied:
How shall we proceed?  Is this PR good as-is or do you suggest some change?

Ansuel replied:
Ok as is. It just got backported. Was just me surprised noticing this and wanted to put some text about it ahhahah

graysky2 replied:
Sounds good, thanks!

robimarko commented on target/linux/qualcommax/config-6.1:
```diff
@@ -28,6 +28,7 @@ CONFIG_ARM64_ERRATUM_2054223=y
 CONFIG_ARM64_ERRATUM_2067961=y
 CONFIG_ARM64_ERRATUM_2077057=y
 CONFIG_ARM64_ERRATUM_2658417=y
+CONFIG_ARM64_ERRATUM_2966298=y
```
Qualcommax is using Cortex-A53 cores, so it shouldn't be affected by this at all

robimarko replied:
Same goes for all targets in OpenWrt, none of them uses A520 core, it probably makes sense to enable it for armsr only though that is targeting ARMv8

graysky2 replied:
At least two checks failed which prompted me to look into the config changes.  I cannot remember which two they were though.  I think one as `armsr/armv8`... I will remove the modifications and see which ones fail.

robimarko replied:
They are just using the default value, for OpenWrt it should just be disabled in the generic config.

Ansuel replied:
(aka setting this as # ... is not defined in the generic configuration)

robimarko replied:
Yes, and then you can drop all of the target config changes as we dont have that core in any OpenWrt target.

graysky2 replied:
Done.

graysky2 committed:
config-6.1: disable CONFIG_ARM64_ERRATUM_2966298

Disable new ksym globally as OpenWrt does not have any targets that
use A520 cores.

Reviewed-by: Robert Marko <robimarko@gmail.com>
Signed-off-by: John Audia <therealgraysky@proton.me>

robimarko commented:
LGTM,
`Reviewed-by: Robert Marko <robimarko@gmail.com>`
